packages/hardhat/contracts/DecentralizedEngergyCommunity.sol

```diff
@@ -0,0 +1,121 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { AccessControl } from "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+
+/**
+ * @title   DecentralizedEnergyCommunity
+ * @notice  This contract is the main contract for the Decentralized Energy Community
+ */
+contract DecentralizedEnergyCommunity is AccessControl {
+	struct Community {
+		uint256 id;
+		string name;
+		Participant[] participants;
+		bool active;
+		bool hasProducer;
```

Comment:
if want to optimize more 'deeply' we could try to use a byte and bitflip if they don't to this optimization for us.

Reply:
since id is now uint32, the bools can be packed into one slot with the id